app/services/credits/applied_coupon_service.rb

```diff
@@ -21,7 +21,12 @@ def create
         amount_currency: applied_coupon.amount_currency,
       )
 
-      applied_coupon.mark_as_terminated! if applied_coupon.coupon.percentage? || credit_amount >= remaining_amount
+      applied_coupon.frequency_duration -= 1 if applied_coupon.recurring?
+      if should_terminate_applied_coupon?(credit_amount)
+        applied_coupon.mark_as_terminated!
+      elsif applied_coupon.recurring?
```

Comment:
a simple `else` is enought or I'm missing something?

Reply:
Here are two cases for updating applied coupon. Either we want to mark termination if certain conditions are met or we just want to save updated frequency_duration. However, frequency_duration should be saved only for recurring coupons. We don't want to open DB connection otherwise...